src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -146,7 +148,7 @@ impl SinkConfig for CloudwatchLogsSinkConfig {
     }
 
     fn input(&self) -> Input {
-        Input::log()
+        Input::new(self.encoding.clone().config().input_type())
```

Comment:
Why the `clone` for these?

Reply:
Asked about this up here: https://github.com/vectordotdev/vector/pull/12561#discussion_r863985134

Reply:
Answer in https://github.com/vectordotdev/vector/pull/12561#discussion_r864198926.

Btw. generally if a `clone` is not necessary, `clippy` would complain. However it can't see through the design decision of making `self` owned or not 😄

Reply:
Adressed in 2c8749009e169f85957abcc94064aa40e4a7849b